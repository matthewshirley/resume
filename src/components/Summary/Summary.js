import React from 'react';
import PropTypes from 'prop-types';

import Box from '../common/Box';
import Text from '../common/Text';
import Block from '../common/Block';

const Summary = ({ summary }) => {
  if (!summary) return (null);

  return (
    <Block title="About">
      <Box paddingBottom={4} borderBottom>
        <Text m={0}>{summary}</Text>
      </Box>
    </Block>
  );
};

Summary.propTypes = {
  summary: PropTypes.string,
};

Summary.defaultProps = {
  summary: null,
};

export default Summary;
