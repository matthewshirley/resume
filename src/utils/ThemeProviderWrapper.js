import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';

/**
 * ThemeProvider for application and test specs
 *
 * @param {object} props
 * @param {element} props.children
 */
const ThemeProviderWrapper = ({ children }) => (
  <ThemeProvider theme={theme}>
    { children }
  </ThemeProvider>
);

ThemeProviderWrapper.propTypes = {
  children: PropTypes.element,
};

ThemeProviderWrapper.defaultProps = {
  children: null,
};

export default ThemeProviderWrapper;
