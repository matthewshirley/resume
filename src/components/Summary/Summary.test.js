import React from 'react';
import renderer from 'react-test-renderer';

import Summary from './Summary';
import ThemeProviderWrapper from '../../utils/ThemeProviderWrapper';


describe('Summary', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProviderWrapper>
          <Summary summary="Test" theme={{ colors: { gray2: '#ffffff' } }} />
        </ThemeProviderWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does not render without a summary', () => {
    const tree = renderer
      .create(<Summary summary={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
