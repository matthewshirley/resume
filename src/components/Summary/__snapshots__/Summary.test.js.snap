// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary does not render without a summary 1`] = `null`;

exports[`Summary renders correctly 1`] = `
<div
  className="sc-AxjAm sc-AxhCb eGofQz"
>
  <div
    className="sc-AxjAm fIbOII"
  >
    <p
      className="sc-AxirZ eKWZht"
      color="primary"
      fontSize={3}
    >
      About
    </p>
  </div>
  <div
    className="sc-AxjAm jBohcl"
  >
    <div
      className="sc-AxjAm kSMmJx"
    >
      <p
        className="sc-AxirZ haCkUz"
        fontSize={1}
      >
        Test
      </p>
    </div>
  </div>
</div>
`;
